Algorithm/ThreadManager.py

```diff
@@ -68,13 +59,37 @@ def return_best_score(cls) -> Tuple[int, dict]:
         """
         :return: Tuple of best result and which set of parameters
         """
-        return sorted(cls.results, key=lambda item: item.score, reverse=False)[0]
+        return sorted(cls.results.values(), key=lambda res: res.score, reverse=False)[0]
+
+
+class ThreadManagerGUI(ThreadManager):
```

Comment:
Does it really necessary to have 2 diff classes ThreadManager?

Reply:
No ,but I divided the class so that we can use it on GUI less runs